app/assets/stylesheets/provider/_links.scss

```diff
@@ -45,18 +47,18 @@ input[type='submit'].action {
     &:hover {
       color: $link-hover-color;
     }
-    // https://github.com/bokmann/font-awesome-rails/blob/master/vendor/assets/stylesheets/font-awesome.css.erb
-    // http://fortawesome.github.com/Font-Awesome/
-    &.undo:before, &.revert:before { @include iconic("\f0e2\0000a0"); }
+    // For reference:
+    // https://github.com/FortAwesome/font-awesome-sass/blob/5.15.1/assets/stylesheets/font-awesome/_variables.scss#L1319
+    &.undo:before, &.revert:before { @include iconic("\f2ea\0000a0"); }
     &.add:before, &.new:before  { @include iconic("\f055\0000a0"); }  // plus-sign
-    &.delete:before   { @include iconic("\f014\0000a0"); }  // trash
-    &.refresh:before   { @include iconic("\f021\0000a0"); } // resend
-    &.edit:before     { @include iconic("\f040\0000a0"); }  // pencil
+    &.delete:before   { @include iconic("\f2ed\0000a0"); }  // trash
+    &.refresh:before   { @include iconic("\f2f1\0000a0"); } // resend
+    &.edit:before     { @include iconic("\f303\0000a0"); }  // pencil
     &.bolt:before     { @include iconic("\f0e7\0000a0"); }  // bolt
     &.fire:before   { @include iconic("\f06d\0000a0"); }    // fire
     &.off:before, &.disable:before, &.suspend:before { @include iconic("\f28b\0000a0"); }     // off
     &.activate:before, &.enable:before, &.resume:before { @include iconic("\f144\0000a0"); }  // unlock
-    &.message:before { @include iconic("\f003\0000a0"); }   // unlock
+    &.message:before { @include iconic("\f0e0\0000a0"); }   // unlock <i class="far fa-envelope"></i>
```

Comment:
Original: https://fontawesome.com/v4/icon/envelope-o

In v5 it's different in `fas` (standard) style: https://fontawesome.com/v5/icons/envelope?f=classic&s=solid

In `far` style it's the same as the previous one: https://fontawesome.com/v5/icons/envelope?f=classic&s=regular

But I don't know how to force this style with just unicode code. Font-weight didn't work.

I think it looks OK though:

before:

<img width="495" height="192" alt="Screenshot From 2025-09-17 13-47-05" src="https://github.com/user-attachments/assets/7b099c6a-c4c6-4bae-a4f9-20194d5d1a3d" />

after:

<img width="495" height="192" alt="Screenshot From 2025-09-17 13-47-09" src="https://github.com/user-attachments/assets/4aafe3b3-91f1-4a72-add0-fa4db6a09d68" />